docs/functions/dialog.md

```diff
@@ -0,0 +1,109 @@
+<!--
+ - SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ - SPDX-License-Identifier: AGPL-3.0-or-later
+-->
+```ts static
+import {
+	spawnDialog,
+} from '@nextcloud/vue/dist/Functions/dialog.js'
+```
+
+## Definitions
+
+```ts static
+/**
+ * Helper to spawn a Vue dialog without having to mount it from a component
+ *
+ * @param dialog The dialog component to spawn - the component must emit the 'close' event whenever it is closed
+ * @param props Properties to pass to the dialog
+ * @param props.container Optionally pass a query selector for the dialog container element
+ * @param onClose Callback when the dialog is closed (parameters of the 'close' event of the dialog)
+ */
+function spawnDialog(
+	dialog: Component | AsyncComponent,
+	props: Record<string, unknown>,
+	onClose: (...rest: unknown[]) => void = () => {},
+): Vue;
+```
+
+## Usage
+
+The main use case is to be able to spawn a dialog from code, without the need of including the dialog component in the template.
+So a Vue dialog can be spawned in any context and not just from Vue components but also from callback functions or other API.
+
+```vue
+<template>
+	<div>
+		<NcButton @click="onSpawnDialog">🖱 click to spawn a dialog</NcButton>
+	</div>
+</template>
+
+<script>
+import { spawnDialog } from '@nextcloud/vue/dist/Functions/dialog.js'
```

Comment:
Current module imports don't work in styleguidist (or we need to configure it somehow). 

This is why we added components/functions to globals like this

https://github.com/nextcloud-libraries/nextcloud-vue/blob/b7eeee8966b33e8f963b30d665b48ada2c2c6d48/styleguide/global.requires.js#L162-L171

Reply:
Hm in local styleguide it works?

Reply:
It shouldn't...

Reply:
I would prefer to stick to globals for now, even if it works another way. Thus, there will be less confusion when migrating to Storybook

Reply:
I would prefer to have explicit imports to have complete examples. 

But they don't work and require additional setup. CI error is relevant.

Reply:
Should be fixed